### PR TITLE
Fix OrdParser

### PR DESCRIPTION
**Pull request by kobby-pentangeli**
# Motivation

- [x] A re-implementation of the `OrdParser`: Closes https://infinityswap.atlassian.net/browse/EPROD-853
- [x] A (re)implementation of `InscriptionId` (IID): Closes https://infinityswap.atlassian.net/browse/EPROD-856

## Changes

- [x] Updated `Brc20Deploy` with the new `self_mint` field
- [x] `Brc20::deploy()` function now takes an additional argument `self_mint: Option<bool>`
- [x] Updated `Nft` with the new fields `rune` and `parents` (previously, it was `parent: Option<Vec<u8>>`; now, it's `parents: Vec<Vec<u8>>`)
- [x] Changed `struct OrdParser` to `enum OrdParser` 
- [x] Changed `NftId` to `InscriptionId`

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [ ] This PR does not affect public APIs
- [ ] This PR does not rely on a new version of external services
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: src/inscription/iid.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+    if hex.is_empty() || hex.len() > 1 {
+        panic!();
+    }
+
+    format!("{}i{n}", hex.repeat(64)).parse().unwrap()
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (F3kilo) [inline src/inscription/iid.rs:156]: I think better to move `set_using` inside `mod tests{}`.

File: src/wallet/parser/envelope.rs [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+    PARENT_TAG, POINTER_TAG, PROTOCOL_ID, RUNE_TAG,
+};
+use crate::Nft;
+
+type ParseResult<T> = std::result::Result<T, ScriptError>;
+type RawEnvelope = Envelope<Vec<Vec<u8>>>;
+pub(crate) type ParsedEnvelope = Envelope<Nft>;
+
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct Envelope<T> {
```
Comment (F3kilo) [inline src/wallet/parser/envelope.rs:21]: Doc comment

File: src/wallet/parser.rs [hunk 3]
```diff
@@ -1,177 +1,354 @@
...
+use bitcoin::Transaction;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use self::envelope::ParsedEnvelope;
+use crate::wallet::RedeemScriptPubkey;
+use crate::{Brc20, Inscription, InscriptionParseError, Nft, OrdError, OrdResult};
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum OrdParser {
```
Comment (F3kilo) [inline src/wallet/parser.rs:13]: Doc comment

**Review by F3kilo — APPROVED** [review]
Great work!
I have only code style suggestions.

File: src/inscription/iid.rs (hunk 1 above)
Comment (kobby-pentangeli) [reply src/inscription/iid.rs]: Done

**Review by veeso — APPROVED** [review]

**Review by F3kilo — APPROVED** [review]

**Review by veeso — APPROVED** [review]

**Pull request merged**